c10/core/impl/PyInterpreter.h

```diff
@@ -17,6 +17,10 @@ struct IValue;
 class OperatorHandle;
 struct TensorImpl;
 struct SafePyObject;
+
+namespace impl {
+struct CUDATraceFunctionWrapper;
+} // namespace impl
```

Comment:
As suggested above, it may be better to fully define this struct here, instead of just forward-declaring it.

Reply:
I believe some forward-declaration will still be needed, even if I put the FunctionWrapper struct in the `PyInterpreter.h` file, because the structs are still circularly dependent on each other. Please correct me if I'm wrong.

Reply:
I saw you had to forward-declare `PyInterpreter` but that seems a much lower price to pay than what we had before?